kiarash/src/index.ts

```diff
@@ -0,0 +1,52 @@
+type Branch = "architecture" | "clean-code" | "languages";
+
+abstract class Concept {
+  constructor(private name: string, protected branch: string) {}
+
+  getName() {
+    return this.name;
+  }
+
+  abstract getBranch(): string;
+}
+
+class devConcept extends Concept {
+  constructor(name: string, branch: Branch) {
+    super(name, branch);
```

Comment:
Set the name as specified in the issue. The constructor should only accept a branch.